src/map/script.cpp

```diff
@@ -19901,7 +19922,16 @@ BUILDIN_FUNC(instance_mapname)
  *------------------------------------------*/
 BUILDIN_FUNC(instance_id)
 {
-	script_pushint(st, script_instancegetid(st));
+	if (script_hasdata(st, 2)) {
+		enum instance_mode mode = static_cast<instance_mode>(script_getnum(st, 2));
+		
+		if (mode < IM_NONE || mode >= IM_MAX) {
+			ShowError("buildin_instance_create: Unknown instance mode %d for '%s'\n", mode, script_getstr(st, 2));
```

Comment:
buildin_instance_create should be buildin_instance_id

Additionally your output is weird. %d mode=getnum(2) and %s for getstr(2). So you treat the same parameter in two ways.